### DISCUSSION TITLE
Missing python modules for SCM tutorial

### DISCUSSION THREAD

**Discussion by timsliwinski-noaa** [NCAR/ccpp-scm#350] [General]
For my most recent project, I'm needing to use the CCPP-SCM to test changes to the physics code. However, since I have little experience running the SCM before, I wanted to walk through the tutorial listed here: https://dtcenter.org/ccpp-scm-online-tutorial

I made it into exercise 1 and ran the preconfigured 6-suite multi-run on the twpice datasets with no issues. However, when I tried to generate the output plots, I got errors regarding missing modules:

Traceback (most recent call last):
  File "/Users/tim.sliwinski/SCM/ccpp-scm/scm/run/./scm_analysis.py", line 11, in <module>
    from configobj import ConfigObj, flatten_errors
ModuleNotFoundError: No module named 'configobj'

I did a grep to search for the configobj module inside the entire repo, but it was not found.

Is the tutorial expected to still be functional or is it no longer supported? 
Is there any way to get the necessary Python modules added back in to the repository to make the tutorial functional again?

**Comment by dustinswales** [comment]
@timsliwinski-noaa Sorry to hear you are running into trouble.
If not running the SCM on a supported platform, there are several python modules needed by the SCM in your python environment.
The following Python modules are required: xarray, dask, netCDF4, bottleneck, matplotlib, f90nml, configobj, shapely, pandas, and scipy.

**Reply by timsliwinski-noaa** [reply]
@dustinswales Thank you for the information! I am running the SCM on a Mac M1 laptop, so, yes, probably outside of the supported platform listing. I appreciate the listing of dependencies though. I did not see them in the documentation (and if I overlooked them, I apologize). I've installed all the dependencies except for bottleneck (seems to have it's own issues with the M1 platform). If I can work around that, I'll give it a try again.

**Reply by timsliwinski-noaa** [reply]
@dustinswales Figured out a workaround for the bottleneck issue on Mac M1 (see note below) and the tutorial plots all worked with the suggested suite of dependencies. Thanks again!

NOTE for future reference: Workaround for issues (e.g. gcc-12: error: unrecognized command-line option '-iwithsysroot/...') installing bottleneck on Mac M1: Use clang as the compiler (CC=clang,CXX=clang), not homebrew's gcc (CC=gcc-12,CXX=g++-12).

**Comment by RobertPincus** [comment]
@dustinswales Maybe y'all want to a conda `environment.yml` and/or PyPi `requirements.txt` file?

**Reply by RobertPincus** [reply]
want to add, I meant to say.

**Reply by dustinswales** [reply]
@RobertPincus Good idea. Thanks.
